tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/Lambda/LambdaCommon.cs

```diff
@@ -141,5 +156,11 @@ internal static void Log(string message, Exception ex = null, bool debug = true)
             Console.WriteLine($"{DateTime.UtcNow:yyyy-MM-dd HH:mm:ss:fff} [DD_TRACE_DOTNET] {message} {ex?.ToString().Replace("\n", "\\n")}");
         }
     }
+
+    internal static string GetApiKey()
+    {
+        var config = new ConfigurationBuilder(GlobalConfigurationSource.Instance, TelemetryFactory.Config);
+        return config.WithKeys(ConfigurationKeys.ApiKey).AsRedactedString();
+    }
```

Comment:
Hmmm, you almost certainly don't want to be calling this _every_ invocation 🤔 It probably would make sense to either make it a lazy initialized value so that you just check it once at most I guess. 

Also, this still will end up doing a _lot_ of work with all the instrumentation etc - maybe that's ok, but you _might_ want to consider moving the bail out earlier up the call stack? Not required though.

Finally, is the API key _always_ set through an environment variable? If so, then you could do something like this instead:

```csharp
private static Lazy<bool> IsApiKeyAvailable = new(()=> IsApiKeyAvailable());
private static bool IsApiKeyAvailable()
    => !string.IsNullOrEmpty(EnvironmentHelpers.GetEnvironmentVariable(ConfigurationKeys.ApiKey));
```

Reply:
Updated to be lazy, and also changed it to still use the configuration, since we could be reading API key from `datadog.yaml`